src/main/java/com/google/cloud/tools/appengine/experimental/AppEngineRequest.java

```diff
@@ -16,15 +16,14 @@
 
 package com.google.cloud.tools.appengine.experimental;
 
-import java.util.concurrent.Future;
+import com.google.cloud.tools.appengine.experimental.internal.cloudsdk.CloudSdkRequestFuture;
+
+import java.io.InputStream;
 
 /**
  * Interface for remote App Engine Requests.
  * @param <T> return request result type
  */
 public interface AppEngineRequest<T> {
```

Comment:
Yeah DeployResult is just a weak wrapper around a string. Potentially for other requests you can return more structured data.

`AppEngineRequestFuture<List<Instances>>` or something might be an example